### PR TITLE
Split up update user endpoint

### PR DESCRIPTION
**Pull request by manishsaha**
## Overview
Prior to this change, the `/user/update` endpoint was incredibly dense and tried to accommodate changing all of the user's fields at once. By separating these endpoints into three different endpoints: `/user/updateDemographics/`, `/user/updateInterests/`, `/user/updateOrganizations/`, we can stratify the various updates a user may request. This will also be easier for iOS and Android when it comes to networking.

part 2/n of the ⭐ clean code⭐ journey

## Changes Made
Lots of map/reduce and promise functions to properly update user's information while gracefully ignoring malformed input.

## Test Coverage
Postman coverage.

## Next Steps
More update endpoints for social media and goals.

### CODE REVIEW

**Review by orkosinha — APPROVED** [review]
Makes sense

File: src/entities/User.ts [hunk 1]
```diff
@@ -42,19 +42,31 @@ class User {
   graduationYear: string | null;
 
   /** User's major */
-  @ManyToOne(type => CornellMajor, major => major.users, { nullable: true, onDelete: "CASCADE" })
+  @ManyToOne(
+    type => CornellMajor,
+    major => major.users,
+    { nullable: true, onDelete: 'CASCADE' }
+  )
```
Comment (leewang-codes) [inline src/entities/User.ts:49]: nice, we should use a repo wide code formatter like prettier if we aren't already

**Review by leewang-codes — APPROVED** [review]
nice work! we love ~ cleanliness ~

File: src/entities/User.ts (hunk 1 above)
Comment (manishsaha) [reply src/entities/User.ts:49]: Yeah, I'm working on a pre-commit hook that will run prettier/tslint automatically for us.

File: src/routers/UpdateDemographicsRouter.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      'graduationYear',
+      'hometown',
+      'major',
+      'profilePictureURL',
+      'pronouns',
+    ];
+
+    /* sanitize fields */
+    for (const key in body) {
+      if (!validFields.includes(key) || key === 'googleID' || key === 'netID') {
```
Comment (yuna878) [inline src/routers/UpdateDemographicsRouter.ts:29]: Doesn't first case take care of second&third cases?

**Review by yuna878 — APPROVED** [review]
Yay for ⭐ clean code⭐ journey

File: src/routers/UpdateDemographicsRouter.ts (hunk 2 above)
Comment (manishsaha) [reply src/routers/UpdateDemographicsRouter.ts:29]: `googleID` and `netID` would be in `validFields`, but we're explicitly stating that they're immutable here.

**Pull request merged**